src/main/java/com/conveyal/analysis/controllers/GtfsVectorTileMaker.java

```diff
@@ -0,0 +1,293 @@
+        return stopsIndex;
+    }
+
+    /** RemovalListener triggered when a spatial index is evicted from the cache. */
+    private void logCacheEviction (String feedId, STRtree value, RemovalCause cause) {
+        LOG.info("Vector tile spatial index removed. Feed {}, cause {}.", feedId, cause);
+    }
+
+    /** Produce a single Mapbox vector tile for the given unique GTFS feed ID and tile coordinates. */
+    public byte[] getTile (String bundleScopedFeedId, int zTile, int xTile, int yTile) {
+
+        // It might make sense to unify these two values into a single compound object.
+        STRtree shapesIndex = shapeIndexCache.get(bundleScopedFeedId);
+        STRtree stopsIndex = stopIndexCache.get(bundleScopedFeedId);
+
+        final long startTimeMs = System.currentTimeMillis();
+        final int tileExtent = 4096; // Standard is 4096, smaller can in theory make tiles more compact
+        Envelope wgsEnvelope = MapTile.wgsEnvelope(zTile, xTile, yTile);
+        Collection<Geometry> patternGeoms = new ArrayList<>(64);
+        for (LineString wgsGeometry : (List<LineString>) shapesIndex.query(wgsEnvelope)) {
+            Geometry tileGeometry = clipScaleAndSimplify(wgsGeometry, wgsEnvelope, tileExtent);
+            if (tileGeometry != null) {
+                patternGeoms.add(tileGeometry);
+            }
+        }
+        JtsLayer patternLayer = new JtsLayer("conveyal:gtfs:patternShapes", patternGeoms, tileExtent);
+
+        Collection<Geometry> stopGeoms = new ArrayList<>(64);
+        for (Stop stop : ((List<Stop>) stopsIndex.query(wgsEnvelope))) {
+            Coordinate wgsStopCoord = new Coordinate(stop.stop_lon, stop.stop_lat);
+            if (!wgsEnvelope.contains(wgsStopCoord)) {
+                continue;
+            }
+
+            Map<String, Object> properties = new HashMap<>();
+            properties.put("feedId", stop.feed_id);
+            properties.put("id", stop.stop_id);
+            properties.put("name", stop.stop_name);
+            properties.put("lat", stop.stop_lat);
+            properties.put("lon", stop.stop_lon);
+
+            // Factor out this duplicate code from clipScaleAndSimplify
+            Coordinate tileCoord = wgsStopCoord.copy();
+            tileCoord.x = ((tileCoord.x - wgsEnvelope.getMinX()) * tileExtent) / wgsEnvelope.getWidth();
+            tileCoord.y = ((wgsEnvelope.getMaxY() - tileCoord.y) * tileExtent) / wgsEnvelope.getHeight();
+
+            Point point =  GeometryUtils.geometryFactory.createPoint(tileCoord);
+            point.setUserData(properties);
+            stopGeoms.add(point);
+        }
+        JtsLayer stopsLayer = new JtsLayer("conveyal:gtfs:stops", stopGeoms, tileExtent);
+
+        // Combine these two layers in a tile
+        JtsMvt mvt = new JtsMvt(patternLayer, stopsLayer);
+        MvtLayerParams mvtLayerParams = new MvtLayerParams(256, tileExtent);
```

Comment:
Should this be moved into a `private static final int TILE_SIZE = 256;`?

Reply:
Yes, we'll add symbolic constants for all magic numbers. This is just left over from prototyping. These also need some comments explaining that they're the usual standard values, and what they do.